_docs/software_support_lifecycle.md

```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title: Software Support Lifecycle
+date: 2020-12-09
+summary: Vox Pupuli policy on supporting upstream software
+---
+
+## Software Support Lifecycle
+
+As a part of Vox Pupuli's mission to provide a home for modules that still have a userbase and need development, we need to define a clear support strategy for what operating systems, dependencies, and upstream software we will support and for how long those will be supported.
+
+As we operate entirely on the graces of our community members being willing to voluntarily contribute to our modules, we also need to be respective of their time and resources and not add undue stress in order to support software that upstream creators no longer support.
+
+To account for this, we no longer support any software or operating system after the date that the maintainers have marked said software or operating system as End of Life. However, this does not mean we will immediately remove code that allows for the module to run on End of Life'd software or operating systems, but the community will no longer be expected to provide support to users for that software.
+
+There are exceptions and the full policy is detailed in the next section.
+
+### Support Lifecycle
+
+#### Operating Systems
+An Operating System is no longer considered supported once it reaches an End of Life date determined by the upstream maintainer* of the operating system. To ensure our users have enough time to adjust, we will take the following steps to ensure as seamless of a transition as possible:
+
+1. A deprecation PR will be created no later that 3 months before an Operating System's End of Life date.
+2. Removal of the End of Life version from the metadata.json and tests to be completed by the End of Life date.
+3. Issues relating to the End of Life operating system will no longer be addressed after the End of Life date.
+4. Code allowing a module to be run on the End of Life operating system will be left in the module until one of the following milestones are reached after which time the code will be removed:
+    1. Maintaining the code prevents the module from moving forward.
+    2. 12 Months after the End of Life date.
```

Comment:
I don't think that this is realistic. It already requires a bunch of time to go through modules and remove an OS from metadata.json, review the tests, get that merged, make a new release. I won't recheck the module a year later to drop dead code. I still believe code related to an EOL OS should be removed when we purge the OS from the metadata.json

Reply:
Agreed. Dead code is bad. It's confusing to new contributors. Having code implies support, but how do you know it actually works? I'm also in favor of a hard cut. Dropping it means dropping it. No grace period where you have Schödinger's support.

Reply:
I agree with @bastelfreak that removing code for unsupported functionality is a good idea. People can just use the version which still supports their old OS or software version.

If we don't tightly couple removing unsupported features to deprecations, it might actually make it more difficult for contributors since they now have to work around this code. For these contributors, it's not going to always be obvious if code is part of the deprecated feature set.

Reply:
If this is the prevailing desire and there isn't anyone who wants to push back, then I'll add this into the document tomorrow.

Reply:
There weren't any new people voting for keeping the code a year, so I guess it's safe to say that the majority is fine with dropping the code in the moment where we drop the OS from metadata.json?

Reply:
Same context @bastelfreak is asking from: does everybody consider PR CI being red as «maintaining the code prevents the module from moving forward»? Technically it's not actually stopping it, but while one might be inclined to check if it's okay for a certain set of a module's CI checks to be failing because they're from an EOL OS, it's a mental load issue.

Sure, it's possible to handle manually parsing if a CI is allowed to be red, but I'd argue that as soon as an EOL OS bumps the CI into the red, it's ripe for removal; at the very least from metadata.json if *only* the OS-specific CI is an issue, and just wholesale otherwise.